modules/javafx.controls/src/main/java/javafx/scene/control/skin/ProgressIndicatorSkin.java

```diff
@@ -541,6 +549,10 @@ public DeterminateIndicator(ProgressIndicator control, ProgressIndicatorSkin s,
             updateProgress(control.getProgress());
         }
 
+        private void unregisterListener() {
+            unregisterChangeListeners(text.fontProperty());
+        }
+
```

Comment:
just a side-note: this will remove the whole chain of listeners to that property. Not a problem here, because the indicator seems to be the only participant listening. A bit strange using the skin's LambdaMultiplePropertyChangeListenerHandler ...

Reply:
Yes, do you know an API for registerChangeListener which allows me to only remove a specific listener?

Reply:
no way, afaik - there had been debates when the api was added somewhere in jbs, don't recall exactly where

Reply:
here's where it was added: https://bugs.openjdk.java.net/browse/JDK-8151617

Reply:
I don't see an API in the discussion, on how to remove a specific listener. Did i miss something? It seems to me, that the API is written to support only a single listener per property.

Reply:
the api is in a related issue (should be noted in the issue above). The use-cases mentioned in the isssue: it's a do-once scenario with the skin (and its subclasses) as the only user: multiple listeners can be registered, typically by a skin and/or its subclasses, each can be pre/postpended (or replaced) in the chain for that property and all are unregistered at dispose time. So I think it's really bad to use it in another collaborator like the indicator, particularly if that is re-created often and each of the instances need to remove the listener.

Reply:
ahh yeah, you are right - somehow missed that text is private to the private indicator